### PR TITLE
build: make libyami works on android

### PR DESCRIPTION
**Pull request by YuJiankang**
make the libyami works on android. verified with v4l2decode and v4l2encode.
Signed-off-by: jkyu jiankang.yu@intel.com

### CODE REVIEW

File: decoder/vaapidecoder_h264.h [hunk 1]
```diff
@@ -47,9 +51,15 @@ class VaapiDecoderH264 : public VaapiDecoderBase {
 
     class DPB {
         typedef VaapiDecoderH264::RefSet RefSet;
+#ifdef ANDROID
```
Comment (xuguangxin) [inline decoder/vaapidecoder_h264.h]: could we define this in  common/Functional.h

File: decoder/vaapidecoder_mpeg2.cpp [hunk 2]
```diff
@@ -206,8 +210,13 @@ VaapiDecoderMPEG2::DPB::getReferencePictures(const PicturePtr& current_picture,
 }
 
 VaapiDecoderMPEG2::VaapiDecoderMPEG2()
+#ifdef ANDROID
```
Comment (xuguangxin) [inline decoder/vaapidecoder_mpeg2.cpp]: use common/Functional.h

File: codecparsers/mpeg2_parser.cpp [hunk 3]
```diff
@@ -83,14 +83,22 @@ namespace MPEG2 {
 
     // default matrix for intra blocks
     // section 6.3.7
+#ifdef ANDROID
+    const static std::array<const uint8_t, 64> kDefaultIntraBlockMatrix
```
Comment (xuguangxin) [inline codecparsers/mpeg2_parser.cpp]: better define this to Arrary and move all things to common/Array.h

File: decoder/Android.mk [hunk 4]
```diff
@@ -22,7 +21,10 @@ LOCAL_SRC_FILES += \
         vaapidecoder_vp9.cpp
 
 LOCAL_SRC_FILES += \
-        vaapidecoder_jpeg.cpp
```
Comment (xuguangxin) [inline decoder/Android.mk:25]: So original code can support jpeg?

File: encoder/Android.mk [hunk 5]
```diff
@@ -12,9 +12,6 @@ LOCAL_SRC_FILES += \
         vaapiencoder_h264.cpp
 
 LOCAL_SRC_FILES += \
-        vaapiencoder_jpeg.cpp
```
Comment (xuguangxin) [inline encoder/Android.mk:15]: why remove jpeg encoder?

**Comment by seanvk** [comment]
@YuJiankang Can you respond to the question of JPEG encoder?  Also this patch is stale.

**Comment by YuJiankang** [comment]
@seanvk , It's only for android, As there is no requirement for android support from customer currently, I  closed this as this patch need some more improvement. For the JPEG encoder, we should support it on android. I will improve the patch and submit it again if we get the final decision of Android related things. thanks a lot.

**Pull request closed without merging**